### ISSUE TITLE
Refactor: rewrite CDC connector validation to Rust

### ISSUE THREAD

**Issue by StrikeW** [risingwavelabs/risingwave#16902]
Currently the validation logic is written in Java, and rely on message passing via protobuf. The interpretation of .proto is different in  Rust and Java, causing burden to maintain the connector.

**Comment by StrikeW** [comment]
For newly supported connector, e.g. #16178 , I suggest we write the validation logic in Rust. @KeXiangWang

**Comment by BugenZhao** [comment]
> The interpretation of .proto is different in Rust and Java, causing burden to maintain the connector.

Can you elaborate more?

**Comment by StrikeW** [comment]
> > The interpretation of .proto is different in Rust and Java, causing burden to maintain the connector.
> 
> Can you elaborate more?

The `oneof` field in Rust can be set to `None`, but after passing to Java it doesn't interpret to `Null`.

**Comment by BugenZhao** [comment]
> The `oneof` field in Rust can be set to `None`, but after passing to Java it doesn't interpret to `Null`.

Do you mean that calling `get` on a mismatched case will return the default value? This should be resolved by checking the "case".

https://protobuf.dev/reference/java/java-generated/#oneof-fields

> `ChoiceCase getChoiceCase()`: Returns the enum indicating which field is set. Returns `CHOICE_NOT_SET` if none of them is set.

**Comment by StrikeW** [comment]
> > The `oneof` field in Rust can be set to `None`, but after passing to Java it doesn't interpret to `Null`.
> 
> Do you mean that calling `get` on a mismatched case will return the default value? This should be resolved by checking the "case".
> 
> https://protobuf.dev/reference/java/java-generated/#oneof-fields
> 
> > `ChoiceCase getChoiceCase()`: Returns the enum indicating which field is set. Returns `CHOICE_NOT_SET` if none of them is set.

Yeah, the different convention causes burden which is unnecessary.

**Comment by StrikeW** [comment]
There is one drawback if we use Rust to write the validate logic: AFAIK Rust doesn't have a unified interface likes the JDBC to access database, to we need to use different Rust driver to interact with different databases. Considering this drawback, it's not really worth the effort to refactor in current stage.

**Comment by xiangjinwu** [comment]
> AFAIK Rust doesn't have a unified interface likes the JDBC to access database

True 😢 It is double-edged rather than a pure drawback in my view. The JDBC unified interface hides certain DB-specific details, leading to us underestimate the complexity in initial implementation and then add continuous patches via its most generic `getObject` and `getColumnTypeName`.

**Comment by BugenZhao** [comment]
> AFAIK Rust doesn't have a unified interface likes the JDBC to access database

What about [sqlx](https://docs.rs/sqlx/latest/sqlx/)? 🥺

**Comment by xiangjinwu** [comment]
> > AFAIK Rust doesn't have a unified interface likes the JDBC to access database
> 
> What about [sqlx](https://docs.rs/sqlx/latest/sqlx/)? 🥺

* It is not as widely-accepted as JDBC as a standard (yet).
* It has no support for MS SQL Server which @KeXiangWang is working on right now. Their future plan is to make it part of SQLx Pro
